### PR TITLE
Feature/eui 1524 &EUI 1525 reinvite user

### PR DESCRIPTION
**Pull request by elviani**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/users/containers/invite-user/invite-user.component.ts [hunk 1]
```diff
@@ -19,50 +22,182 @@ import {AppConstants} from '../../../app/app.constants';
...
+  public handleError(store: Store<any>, errorNumber: number) {
+    const globalError = this.getGlobalError(errorNumber);
+    store.dispatch(new fromAppStore.AddGlobalError(globalError));
+    store.dispatch(new fromAppStore.Go({ path: ['service-down'] }));
+  }
+
+  public getGlobalError(error: number): GlobalError {
+    switch (error) {
+      case 400:
+        return this.get400Error();
```
Comment (paddybasi) [inline src/users/containers/invite-user/invite-user.component.ts:89]: Not overly keen on this approach, as it means we'd have to have more functions per status code etc. How about a mapping instead?

File: src/app/components/service-down/service-down.component.html [hunk 2]
```diff
@@ -1,8 +1,10 @@
 <main class="govuk-main-wrapper govuk-main-wrapper--l" id="main-content" role="main">
   <div class="govuk-grid-row">
     <div class="govuk-grid-column-two-thirds">
-      <h1 class="govuk-heading-xl">Sorry, there is a problem with the service</h1>
-      <p class="govuk-body">Try again later.</p>
+      <h1 class="govuk-heading-xl">{{(currentError)?.header}}</h1>
```
Comment (paddybasi) [inline src/app/components/service-down/service-down.component.html:4]: this might introduce empty h1, not sure if you want an ngIf etc?

File: src/app/components/service-down/service-down.component.html [hunk 3]
```diff
@@ -1,8 +1,10 @@
...
+      <p class="govuk-body" *ngFor="let error of currentError.errors">
+        <a *ngIf="error.urlText" href="{{error.url}}">{{error.urlText}}</a> {{ error.bodyText }}
```
Comment (elviani) [inline src/app/components/service-down/service-down.component.html:6]: might want to use routerLink instead of href

File: src/shared/components/hmcts-error-summary/hmcts-error-summary.component.html [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<div *ngIf="!showWarningMessage" id="errorSummary" class="govuk-error-summary" aria-labelledby="error-summary-title" role="alert" tabindex="-1"
```
Comment (elviani) [inline src/shared/components/hmcts-error-summary/hmcts-error-summary.component.html:1]: there is hmcts-error-summary inside the common-lib already, can we start using that?
I have actually remove all the generic components in manage org and move it to common-lib, but the PR in manage org never been merged and now outdated. but may be a good practice to start using the common-lib for now. (and modify the components in common-lib).

File: src/shared/components/hmcts-main-wrapper/hmcts-main-wrapper.component.html [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<a *ngIf="backLink && !backAction" [routerLink]="backLink" class="govuk-back-link">Back</a>
```
Comment (elviani) [inline src/shared/components/hmcts-main-wrapper/hmcts-main-wrapper.component.html:1]: the same with hmcts-main-wrapper, it is available in common-lib

File: src/app/components/service-down/service-down.component.html (hunk 3 above)
Comment (udaydenduluri33) [reply src/app/components/service-down/service-down.component.html:6]: Done

File: src/users/containers/invite-user/invite-user.component.ts (hunk 1 above)
Comment (udaydenduluri33) [reply src/users/containers/invite-user/invite-user.component.ts:89]: Done

File: src/app/components/service-down/service-down.component.html (hunk 2 above)
Comment (udaydenduluri33) [reply src/app/components/service-down/service-down.component.html:4]: I removed the nullable now. It always should have a error message [it has a default value] I have reused the existing component. if there is no error message default error will be shown. Thanks

File: src/shared/components/hmcts-error-summary/hmcts-error-summary.component.html (hunk 4 above)
Comment (udaydenduluri33) [reply src/shared/components/hmcts-error-summary/hmcts-error-summary.component.html:1]: @elviani I have changed it slightly for handling scenario 429. I will do that later if you alright

File: src/shared/components/hmcts-main-wrapper/hmcts-main-wrapper.component.html (hunk 5 above)
Comment (udaydenduluri33) [reply src/shared/components/hmcts-main-wrapper/hmcts-main-wrapper.component.html:1]: Again @elviani I have changed it slightly. I will do that later if you alright

File: api/inviteUser/inviteUserUtil.spec.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { expect } from 'chai'
+import { getInviteUserUrl } from './inviteUserUtil'
+
+describe('terms And Conditions ', () => {
```
Comment (paddybasi) [inline api/inviteUser/inviteUserUtil.spec.ts:4]: what does this have to do with T&C's? lol

File: api/lib/payloadBuilder.spec.ts [hunk 7]
```diff
@@ -91,7 +91,7 @@ describe('Payload builder', () => {
         expect(organsiationPayload.contactInformation[0].county).to.equal(STATE_VALUES.county)
     })
 
-    it('Should take the postcode and set it as the postcode on the payload.', () => {
+    xit('Should take the postcode and set it as the postcode on the payload.', () => {
```
Comment (paddybasi) [inline api/lib/payloadBuilder.spec.ts:94]: can we please try fixing unit tests rather than commenting out any failures?

File: api/inviteUser/inviteUserUtil.spec.ts (hunk 6 above)
Comment (udaydenduluri33) [reply api/inviteUser/inviteUserUtil.spec.ts]: lol. fixed it now.

File: api/lib/payloadBuilder.spec.ts (hunk 7 above)
Comment (udaydenduluri33) [reply api/lib/payloadBuilder.spec.ts]: Fixed it. Thanks
